lib/temporalio/worker/activity_worker.rb

```diff
@@ -37,8 +38,19 @@ def run(reactor)
       rescue Temporalio::Bridge::Error::WorkerShutdown
         # No need to re-raise this error, it's a part of a normal shutdown
       ensure
-        reactor.async do
+        reactor.async do |async_task|
+          cancelation_task =
+            if graceful_timeout
+              async_task.async do
+                sleep graceful_timeout
+                running_activities.each_value do |activity_runner|
+                  activity_runner.cancel('Worker is shutting down', by_request: false)
+                end
+              end
+            end
+
           outstanding_tasks.each(&:wait)
+          cancelation_task&.stop # all tasks completed, stop cancellations
```

Comment:
Is this required?

Reply:
Yep, since `Async` waits for all outstanding tasks to finish before it releases the control. Without it we would be waiting for graceful timeout every time even if there's no running activities